backend/backend.go

```diff
@@ -122,3 +123,63 @@ func UnmarshalHistoryEvent(bytes []byte) (*HistoryEvent, error) {
 	}
 	return e, nil
 }
+
+func purgeOrchestrationState(ctx context.Context, be Backend, iid api.InstanceID, recursive bool) (int, error) {
```

Comment:
It looks like we never use the `recursive` value here. I'm curious how the tests which do a non-recursive purge are passing?

Reply:
Oh that was a bug. The test was passing because only recursive scenario was being tested there, again a bug. Fixed that. Thanks for catching this

Reply:
Done